netty/src/main/java/io/grpc/netty/GrpcSslContexts.java

```diff
@@ -66,19 +77,19 @@ private GrpcSslContexts() {}
       Protocol.ALPN,
       SelectorFailureBehavior.NO_ADVERTISE,
       SelectedListenerFailureBehavior.ACCEPT,
-      singletonList(HTTP2_VERSION));
+      NEXT_PROTOCOL_VERSIONS);
 
   private static ApplicationProtocolConfig NPN = new ApplicationProtocolConfig(
       Protocol.NPN,
       SelectorFailureBehavior.NO_ADVERTISE,
       SelectedListenerFailureBehavior.ACCEPT,
-      singletonList(HTTP2_VERSION));
+      NEXT_PROTOCOL_VERSIONS);
```

Comment:
I don't know of anything preventing us using NPN for the negotiation, the same rule for protocol selection preference applies (but done by client instead of server). Given NPN is a fallback in grpc-java based on runtime limitations (and not officially supported in HTTP/2 or gRPC), I'm inclined to leave this there unless you feel it really doesn't belong.